pkg/insightstore/filestore.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2020 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package insightstore
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/pipe-cd/pipe/pkg/filestore"
+	"github.com/pipe-cd/pipe/pkg/model"
+)
+
+type Store struct {
+	filestore filestore.Store
+}
```

Comment:
Maybe we should simplify the store interface as below:

```
func (s *Store) LoadChunks(ctx context.Context, projectID, appID string, ...) ([]Chunk, error)
func (s *Store) PutChunk(ctx context.Context, chunk Chunk) error
```

And chunk has a function to get `[]model.Datapoint` (Instead of adding a `List` function to the Store)

Reply:
I meant the store is only for loading/saving the chunks, no more.
Extracting the `model.DataPoint` should be a part of the chunk.
For example,

```
type Chunks []Chunk

func (cs Chunks) DataPoints(from time.Time, count int) ([]*model.DataPoint, error) 
```